srcpkgs/mold/patches/00-makefile.patch

```diff
@@ -0,0 +1,84 @@
+diff --git a/Makefile b/Makefile
+index 207f54c..e20a55d 100644
+--- a/Makefile
++++ b/Makefile
+@@ -1,15 +1,14 @@
+ CC = clang
+ CXX = clang++
+ 
+-MIMALLOC_LIB = mimalloc/out/release/libmimalloc.a
+-GIT_HASH = $(shell [ -d .git ] && git rev-parse HEAD)
++GIT_HASH ?= $(shell [ -d .git ] && git rev-parse HEAD)
+ 
+ CPPFLAGS = -g -Imimalloc/include -pthread -std=c++20 \
+            -Wno-deprecated-volatile \
+            -DMOLD_VERSION=\"0.9.1\" \
+            -DGIT_HASH=\"$(GIT_HASH)\" \
+ 	   $(EXTRA_CPPFLAGS)
+-LDFLAGS = $(EXTRA_LDFLAGS)
++LDFLAGS += $(EXTRA_LDFLAGS)
+ LIBS = -Wl,-as-needed -lcrypto -pthread -ltbb -lz -lxxhash -ldl
+ OBJS = main.o object_file.o input_sections.o output_chunks.o \
+        mapfile.o perf.o linker_script.o archive_file.o output_file.o \
+@@ -41,7 +40,12 @@ else
+   # By default, we want to use mimalloc as a memory allocator.
+   # Since replacing the standard malloc is not compatible with ASAN,
+   # we do that only when ASAN is not enabled.
+-  LDFLAGS += -Wl,-whole-archive $(MIMALLOC_LIB) -Wl,-no-whole-archive
++  ifdef SYSTEM_MIMALLOC
```

Comment:
This looks like it should be upstreamed as well.

Reply:
It has been upstreamed already: https://github.com/rui314/mold/pull/62, https://github.com/rui314/mold/pull/64. But no new releases made with this patch yet.

Reply:
@ericonr let's wait for a release: https://github.com/rui314/mold/issues/65